src/lib/ngsi/ProvidingApplication.h

```diff
@@ -34,16 +34,31 @@
 
 /* ****************************************************************************
 *
-* ProvidingApplication - 
+* ProviderFormat -
+*/
+typedef enum ProviderFormat
+{
+  PfNone,
+  PfJson,
+  PfV2
+} ProviderFormat;
+
+
+
+/* ****************************************************************************
+*
+* ProvidingApplication -
 */
 typedef struct ProvidingApplication
 {
-  std::string   string;
-  MimeType      mimeType;     // Not part of NGSI itself, used by the CB to specify the preferred Mime-Type for CPr interaction
+  std::string     string;
+  MimeType        mimeType;        // Not part of NGSI itself, used by the CB to specify the preferred Mime-Type for CPr interaction
+  ProviderFormat  providerFormat;  // PfJson ("JSON" in mongo): NGSIv1, PfV2: NGSIv2
```

Comment:
`providerFormat` and `mimeType` should be unified (in fact, `mimeType` seems to be a legacy from the XML days) but in a separate PR (in this PR with a `// FIXME PR: ...` mark is enough).

Reply:
Fixed in 2ca5e66 (only a FIXME)

Reply:
I think the moment is (actually, was) "when XML is removed" (because NGSIv1 forwarding also is based in application/json as MIME type), but it's fine in the case you don't want to bother with a little adjust in the comment text.

Reply:
Not sure we ever did forwarding in both JSON and XML.
Changing the text for something broader:
```
// FIXME: Remove 'mimeType'
```
Because saying that is should be removed "when" something happens that has already happened and many years ago ...

Reply:
What about?

```
// FIXME: Remove mimeType as it is useless (we always use application/json to forward, no matter NGSIv1 or NGSIv2)
```

Reply:
Done